### PR TITLE
feat: server variables must have default value

### PR DESCRIPTION
**Pull request by Andras-Csanyi**
This PR brings the capability of checking whether
a server variable has a default value or not. If not,
then we are going to display warning.

The code snippet below will display warning.
```
servers:
  - url: 'https://example.com'
    description: 'A server which has variable without default value'
    variables:
      name:
        default: ''
```

### CODE REVIEW

**Review by dpopp07 — APPROVED** [review]
Looks good to me! 👍 

Let's wait for @barrett-schonefeld to check it out too, before merging

File: src/spectral/rulesets/.defaultsForSpectral.yaml [hunk 1]
```diff
@@ -232,3 +232,15 @@ rules:
     - $.paths[*][*].requestBody.content[*].schema
     then:
       function: string-boundary
+
+  # ensure server variable has default value
+  server-variable-default-value:
+    description: 'Server variable should have default value'
+    message: "{{error}}"
```
Comment (barrett-schonefeld) [inline src/spectral/rulesets/.defaultsForSpectral.yaml:239]: `{{error}}` targets the error message from a custom function.

Here, we want to use the description as the error message.

```suggestion
    message: "{{description}}"
```

File: test/spectral/tests/custom-rules/servers-variable-default-value.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          }
+        }
+      ],
+      paths: {}
+    };
+    const res = await inCodeValidator(spec, true);
+    const result = res.warnings.find(
+      elem => elem.rule === 'server-variable-default-value'
+    );
+    expect(result).not.toBeUndefined();
```
Comment (barrett-schonefeld) [inline test/spectral/tests/custom-rules/servers-variable-default-value.test.js:75]: We should check that the resulting error has the correct error message and path here.

**Review by barrett-schonefeld — CHANGES_REQUESTED** [review]
A couple of small comments, and should be ready to merge after those changes!

File: test/spectral/tests/custom-rules/servers-variable-default-value.test.js [hunk 3]
```diff
@@ -69,9 +69,66 @@ describe('spectral - test whether servers variables have default value', () => {
...
+      ],
+      paths: {}
+    };
+    const res = await inCodeValidator(spec, true);
+    const errorMessageFilter = res.warnings.filter(
+      elem => elem.message === 'Server variable should have default value'
+    );
+    expect(errorMessageFilter.length).toBe(2);
+
+    let pathHits = 0;
```
Comment (barrett-schonefeld) [inline test/spectral/tests/custom-rules/servers-variable-default-value.test.js:113]: Sorry to be picky here! I think it's simpler to utilize the jest tooling here:

```suggestion
    expect(res.warnings[0].path.toEqual(['servers', '0', 'variables', 'name', 'default'])
    expect(res.warnings[1].path.toEqual(['servers', '1', 'variables', 'address', 'default'])
```

Comment (Andras-Csanyi) [reply test/spectral/tests/custom-rules/servers-variable-default-value.test.js:113]: Ahh, awesome!

**Pull request merged**